src/HealthGPS/risk_factor_adjustable_model.cpp

```diff
@@ -44,7 +46,8 @@ double RiskFactorAdjustableModel::get_expected(RuntimeContext &context, core::Ge
     // Apply optional trend to expected value.
     if (apply_trend) {
         int elapsed_time = context.time_now() - context.start_time();
-        expected *= pow(expected_trend_->at(factor), elapsed_time);
+        int t = std::min(elapsed_time, get_trend_steps(factor));
+        expected *= pow(expected_trend_->at(factor), t);
```

Comment:
Could you explain the logic of this bit? It's puzzling me.

Reply:
Sure. `t` is the smaller of the number of elapsed years and the upper bound on elapsed years, used to bound the exponential growth. Maybe `num_trend_steps` is a better name.

So at the start, when `elapsed <= trend_steps`, we have `expect_new = expect * trend^elapsed`, but after we have `expect * trend^trend_steps`.